nncf/experimental/onnx/graph/onnx_graph.py

```diff
@@ -202,24 +205,83 @@ def get_weight_tensor(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
+                    if metatype in [ONNXIdentityMetatype, ONNXQuantizeLinearMetatype]:
+                        return self._get_tensor_from_zero_input(current_node)
+                    if metatype == ONNXReshapeMetatype:
+                        return self._get_weight_tensor_with_reshape(current_node)
+                    weight_port_id = self.get_weight_tensor_port_id(current_node)
+                    if weight_port_id is None:
+                        # The node does not have weight
+                        continue
+                    weight_input = self.get_node_edge_names(current_node.name)['input'][weight_port_id]
+                    return self.get_initializer(weight_input).name, self.get_initializers_value(weight_input)
+                except RuntimeError:
+                    continue
+            raise RuntimeError('Could not find the weight value of the node') from exc
+
+    def _get_param_from_weight_definitions(self, node: onnx.NodeProto, parameter: str) -> int:
+        metatype = ONNX_OPERATION_METATYPES.get_operator_metatype_by_op_name(node.op_type)
+        if metatype in WEIGHT_LAYER_METATYPES:
+            parameter = getattr(metatype.weight_definitions, parameter)
+            if parameter is not None:
+                return parameter
+            raise RuntimeError(f'The metatype {metatype} does not have {parameter} attribute')
+        raise RuntimeError(f'The metatype {metatype} does not belong to a list of metatypes with a weight tensor.')
+
+    def get_weight_tensor_port_id(self, node: onnx.NodeProto) -> int:
+        """
+        Returns input port id, where a weight tensor should output.
+
+        :param node: Node, for which input port id is returned,
+        :return: input port id, where a weight tensor should output.
+        """
+        return self._get_param_from_weight_definitions(node, 'weight_port_id')
+
+    def get_channel_axis(self, node: onnx.NodeProto) -> int:
+        """
+        Returns channel axis for per-channel quantization.
+
+        :param node: Node, for which channel axis id is returned,
+        :return: Channel axis for per-channel quantization.
+        """
+        axis = self._get_param_from_weight_definitions(node, 'channel_axis')
+        if axis is None:
+            raise RuntimeError(f'Could not find the port_id for the node {node.name}')
+        return self._get_param_from_weight_definitions(node, 'channel_axis')
+
+    def get_bias_tensor_port_id(self, node: onnx.NodeProto) -> int:
+        """
+        Returns input port id, where a bias tensor should output.
+
+        :param node: Node, for which input port id is returned,
+        :return: input port id, where a weight bias should output.
+        """
+        return self._get_param_from_weight_definitions(node, 'bias_port_id')
+
+    def _get_weight_tensor_with_reshape(self, node: onnx.NodeProto) -> Tuple[str, np.ndarray]:
```

Comment:
No description.

Reply:
Although I need to remind you that the docstring is not strictly necessary for private functions, in this case I think it might help with understanding this and the other functions' purpose.

Reply:
Done